packages/sdk/src/evm/contracts/smart-contract.ts

```diff
@@ -407,6 +407,17 @@ export class SmartContract<
     this.contractWrapper.updateSignerOrProvider(network);
   }
 
+  async onAbiUpdated(updatedAbi: ContractInterface): Promise<void> {
```

Comment:
you don't need this at this level i think, i would put the `updateAbi()` function directly inside `contractWrapper` and make `this.abi = this.contractWrapper.abi` so it always reflects what's in contractWrapper. That way you don't need it to do callback style since you have this.contractWrapper everyhwere